Add proxy and user_agent labels to user login counter metric

vapopov opened the pull request:
In this PR, added a user agent to the client tools request header. During login, it is passed through the proxy to the auth service. Additionally, information about the public proxy address is included so it can be recorded in metrics.

The `user_login_total` counter now has two labels: proxy and user_agent.
- The `proxy_group_id` label allows us to track which regions are used by customers during login.
- The `user_agent_type` and `version` label helps identify the client tools version in use, to detect any affected versions in future, and monitor the progress of client tools managed updates.

Example:
```
# HELP user_login_per_client Number of times there was a user login with specific client version and GroupID that routes request
# TYPE user_login_per_client counter
user_login_per_client{proxy_group_id="proxy1",user_agent_type="web",version="18.2.5"} 1
user_login_per_client{proxy_group_id="proxy1",user_agent_type="api",version="18.2.5"} 1
user_login_per_client{proxy_group_id="proxy2",user_agent_type="api",version="18.2.5"} 2
```


Related: 
- https://github.com/gravitational/cloud/issues/14225
- https://github.com/gravitational/roundtrip/pull/32

vapopov committed:
Added user login counter metrics labels with proxy information and user-agent version

rosstimothy reviewed (commented):
Agree with Tiago, this seems like it could be  a very high cardinality metric. cc @evanfreed.

sclevine reviewed (commented):
I think the concern about high cardinality on the user agent is worth addressing for scalability reasons, even if authenticated users could likely already cause trouble by connecting many agent versions (which we already record).

Could we parse the user agent for Teleport major version, and only accept reasonable values?

vapopov commented on lib/auth/sessions.go:
```diff
@@ -344,7 +348,11 @@ func (a *Server) newWebSession(
-	UserLoginCount.Inc()
+
+	UserLoginCount.With(prometheus.Labels{
+		teleport.TagUserAgent: fmt.Sprintf("web/%s", teleport.Version),
```
Yes this is web app version (identical to the version of the server), added as optional here

vapopov commented:
> How can we ensure that an attacker doesn't break cloud monitoring by spamming tons of agents and causing Prometheus to fail? Are we going to drop them in cloud?

@tigrato in order to do that, you need a user from the cluster and must complete the login process. We would probably need to reset counters periodically as well

vapopov committed:
Fixed defining default user-agent when `extraHeaders` map was nil
Replaced Sprintf with string concatenation

vapopov committed:
Merge remote-tracking branch 'origin/master' into vapopov/user-login-agent-proxy-tags-metric

vapopov committed:
Fix the env proxy group ID

vapopov committed:
Add periodic clean up and change label for the metric

vapopov commented:
I’ve added a periodic cleanup (hourly) and implemented the label separation by type and version as proposed by @evanfreed.
Since the metric is periodically reset, I’ve introduced a separate metric, `user_login_per_client`, to avoid breaking existing dashboards built on `user_login_total`.

vapopov committed:
Use `ProxyGroupID` from the proxy configuration.
Login counter metric should not be exported.

vapopov committed:
Namespace new metric
CR changes

vapopov commented on lib/auth/auth.go:
```diff
@@ -981,6 +981,17 @@ var (
 	)
+	userLoginCountPerClient = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: teleport.MetricUserLoginPerClientCount,
```
@rosstimothy I’ve added this commit: https://github.com/gravitational/teleport/pull/60239/commits/bdadb18ea42efc2ef830f50bfe6a2c6939677e74, I’m not sure about the subsystem since we can't add it to an existing metric group (to avoid breaking existing dashboards)

vapopov committed:
Removed global constants and changed help message

vapopov committed:
Comments adjustment
